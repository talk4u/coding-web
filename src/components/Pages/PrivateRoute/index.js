import React from 'react'
import {connect} from "react-redux";
import {Route, Redirect} from 'react-router-dom'

const mapStateToProps = (state, ownProps) => {
    return {
        isAuthenticated: state.authReducer.isAuthenticated
    }
}

export const PrivateRouteView = ({component: Component, isAuthenticated, ...rest}) => {
    return (
        <Route
            {...rest}
            render={props=>
                isAuthenticated ? (
                    <Component {...props}/>
                ):(
                    <Redirect
                        to={{pathname: "/sign",
                            state: {from: props.location}
                        }}
                    />
                )
            }
        />
    )

}
const PrivateRoute = connect(
    mapStateToProps,
    null
)(PrivateRouteView)

export default PrivateRoute
