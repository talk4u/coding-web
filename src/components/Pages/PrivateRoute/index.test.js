import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PrivateRoute, {PrivateRouteView} from './index'
import configureStore from "redux-mock-store";
import renderer from 'react-test-renderer';
import {Redirect} from "react-router-dom";

Enzyme.configure({ adapter: new Adapter() });

describe('Private Route Shallow render', () => {
    const initialState = {
        authReducer:{
            isAuthenticated: true
        }
    }
    const mockStore = configureStore();
    let store, container, dumb;
    store = mockStore(initialState)
    describe('with auth True', () => {
        beforeEach(()=>{
            container = shallow(<PrivateRoute store={store}/> );
            dumb = shallow(<PrivateRouteView isAuthenticated={true} component={()=>(<div>hello</div>)}/>)
        });
        it('+++ render the connected(SMART) component', () => {
            expect(container.length).toEqual(1)
        });
        it('+++ render the DUMB component as True', () => {
            expect(dumb.length).toEqual(1)
            expect(dumb.prop('render')().type()).toEqual(<div>hello</div>)
        });
    })
    describe('with auth False', () => {
        beforeEach(()=>{
            dumb = shallow(<PrivateRouteView isAuthenticated={false}/>)
        });
        it('+++ render the DUMB component as False', () => {
            expect(dumb.length).toEqual(1)
            expect(dumb.prop('render')({location:'/'}).type).toEqual(Redirect)
        });
    })

})