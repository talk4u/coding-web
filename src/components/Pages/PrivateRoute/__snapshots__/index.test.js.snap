// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Private Route Shallow render with auth False +++ render the DUMB component as False 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PrivateRouteView
    isAuthenticated={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Private Route Shallow render with auth True +++ render the DUMB component as True 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PrivateRouteView
    component={[Function]}
    isAuthenticated={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
